pkg/operator/ceph/cluster/mgr/spec.go

```diff
@@ -49,6 +49,21 @@ func (c *Cluster) makeDeployment(mgrConfig *mgrConfig) (*apps.Deployment, error)
 		volumes = append(volumes, adminKeyringVol)
 	}
 
+	if c.spec.Mgr.Count > 1 {
+		// create some empty volumes needed by the mgr sidecar
+		volumes = append(volumes,
+			v1.Volume{
+				Name: "rook-config",
+				VolumeSource: v1.VolumeSource{
+					EmptyDir: &v1.EmptyDirVolumeSource{},
```

Comment:
if reconciled it will turn the directory again to empty?

And it just refers to `EmptyDir`? what is privileged here that creates it? I see it just refers to it

Reply:
@parth-gr Not sure about the reconcile part as I haven't seen any error (if the `ceph.conf` disappears I should have seen errors after a while). The code just create empty dirs on the deployment (when mgr count is > 1) this way the container can create files on them without being privileged. Not sure if this answers your question or not?

@travisn is my understanding correct? how can I "force" a reconcile?

Reply:
The emptydir has the lifetime of the pod. If the pod is deleted, then the emptydir will again be created fresh and we will regenerate the config files there. So to test this, you can delete the mgr pod and see it re-created successfully.

Reply:
@travisn I have already tested that and also tested the upgrade and both cases work without issues :+1:

Reply:
@rkachach I am just thinking this would be recreated with every operator restart or ceph crd reconcile, which might be fine or does it will make the data wipe out which might be required?

Reply:
@parth-gr I think the directories are re-created with every pod re-creation (not restart). Operator restarts shouldn't affect the already running pods AFAIK. Anyway, the changes on this PR doesn't change the previous behavior in that sense: it means, we are just creating empty dirs for the pod to avoid requiring privileges in order to create them.